internal/attestation/vsa.go

```diff
@@ -0,0 +1,125 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package attestation
+
+import (
+	"strings"
+	"time"
+
+	"github.com/enterprise-contract/ec-cli/internal/evaluator"
+	"github.com/enterprise-contract/ec-cli/internal/policy"
+	"github.com/enterprise-contract/ec-cli/internal/policy/source"
+	"github.com/in-toto/in-toto-golang/in_toto"
+)
+
+const (
+	// Make it visible elsewhere
+	PredicateVSAProvenance = "https://slsa.dev/verification_summary/v1"
+	StatmentVSA            = "https://in-toto.io/Statement/v1"
+)
+
+type ProvenanceStatementVSA struct {
+	in_toto.StatementHeader
+	Predicate predicate `json:"predicate"`
+}
+
+type policySource struct {
+	uri    string
+	digest map[string]string
+}
+
+type attestationSource struct {
+	uri    string
+	digest map[string]string
+}
+
+type predicate struct {
+	Verifier            map[string]string   `json:"verifier"`
+	TimeVerified        string              `json:"timeVerified"`
+	ResourceUri         string              `json:"resourceUri"`
+	Policies            []policySource      `json:"policies"`
+	InputAttestations   []attestationSource `json:"intputAttestations"`
+	VerificationResult  string              `json:"verificationResult"`
+	VerifiedRules       []string            `json:"verifiedRules"`
+	VerifiedCollections []string            `json:"verfiedCollection"`
+	SlsaVersion         string              `json:"slsaVersion"`
+}
+
+func VsaFromImageValidation(results []evaluator.Outcome, policies []source.PolicySource, policy policy.Policy, attestations []Attestation) (ProvenanceStatementVSA, error) {
+	var verifiedPolicies []policySource
+	for _, p := range policies {
+		verifiedPolicies = append(verifiedPolicies, policySource{uri: p.PolicyUrl()})
+	}
+
+	var verfiedResults int
+	var verifiedLevels []string
+	for _, res := range results {
+		for _, success := range res.Successes {
+			verifiedLevels = append(verifiedLevels, success.Metadata["code"].(string))
+		}
+		verfiedResults = verfiedResults + len(res.Failures)
+	}
+	verificationResult := "Success"
+	if verfiedResults > 0 {
+		verificationResult = "Failure"
+	}
+
+	var verifiedCollections []string
+	for _, source := range policy.Spec().Sources {
+		for _, include := range source.Config.Include {
+			splitInclude := strings.Split(include, "@")
+			if len(splitInclude) > 1 {
+				verifiedCollections = append(verifiedCollections, splitInclude[1])
+			}
+		}
+	}
+
+	var slsaVersion string
+	var digest map[string]string
+	var subject []in_toto.Subject
+	for _, sp := range attestations {
+		slsaVersion = sp.PredicateType()
+		digest = sp.Digest()
+		subject = sp.Subject()
+	}
+
+	return ProvenanceStatementVSA{
+		StatementHeader: in_toto.StatementHeader{
+			Type:          StatmentVSA,
+			PredicateType: PredicateVSAProvenance,
```

Comment:
So we're claiming that this is a predicate of type `https://slsa.dev/verification_summary/v1`. However, there are various attributes here that are not part of that schema, e.g. policies, verifiedRules, verifiedCollections, etc. We would be doing a disservice to our users by taking this approach.

To me that's a clear sign that we should be producing something else...

Reply:
I've changed it to `"https://enterprisecontract.dev//verification_summary/v1"`